src/core/tsi/s2a/handshaker/s2a_handshaker_client.h

```diff
@@ -186,10 +213,17 @@ class S2AHandshakerClient {
   gpr_mu mu_;
   /** This status indicates whether the handshaker call's RECV_STATUS_ON_CLIENT
    *  op is done. **/
-  bool receive_status_finished_;
+  bool receive_status_finished_ = false;
   /** If this field is not nullptr, then it contains arguments needed to
    *  complete a TSI next callback. **/
-  s2a_recv_message_result* pending_recv_message_result_;
+  s2a_recv_message_result* pending_recv_message_result_ = nullptr;
+  /** This variable should be set to true iff the S2A handshaker client instance
+   *  is instantiated as part of a test. **/
+  bool is_test_ = false;
```

Comment:
Can we consolidate these two variables? `is_test_` and `no_calls_for_testing_`

Reply:
I kept them separate for the following reason: there are basic unit tests in `s2a_handshaker_client_test.cc`, and then unit tests for the methods that call the S2A service in `s2a_handshaker_client_caller_test.cc`. 

The variable `is_test_` is set to `true` in both cases (and in certain TSI handshaker tests). The variable `no_calls_for_testing_` is set to `true` in the former case, and `false` in the latter case.

Note that only `is_test_` must be specified in the constructor of the `S2AHandshakerClient`.

What are your thoughts on this strategy?

Reply:
IIUC, `MakeGrpcCall` is not actually called in both cases since `grpc_caller_` callback (which talks to the handshaker server) is not called so not sure why we treat those two tests differently.

Reply:
I did not see this comment is addressed.

Reply:
Sorry Yihua I missed this comment on my last pass. The issue is the following: for the tests in `s2a_handshaker_client_test.cc`, there is no `grpc_caller_` set and, if the `no_calls_for_testing_` variable is not present in `MakeGrpcCallUtil, then `MakeGrpcCall` (and hence `grpc_caller_`) is called. This leads to an error in the test.

After looking at this more carefully, I think it makes sense to do the following:
1. Remove the `s2a_handshaker_client_test.cc` file entirely and add any edge cases that it tests to `s2a_handshaker_client_caller_test.cc`.
2. Remove the `no_calls_for_testing_` variable.

WDYT?

Reply:
The plan (removing the boolean flag and combining these two tests) SGTM.

Reply:
Sounds good. Made these changes in the next commit.